esti/s3_gateway_test.go

```diff
@@ -143,21 +174,23 @@ func TestS3UploadAndDownload(t *testing.T) {
 				objects = make(chan Object, parallelism*2)
 			)
 
-			client := newMinioClient(t, sig.GetCredentials)
+			// Using a minio client only to cover the V2 signing scenario. As V2 signing is already deprecated in AWS it is no longer
+			// supported in its SDK. We should consider removing this test case altogether at some point
+			clt := newMinioClient(t, sig.GetCredentials)
 			wg.Add(parallelism)
 			for i := 0; i < parallelism; i++ {
 				go func() {
 					defer wg.Done()
 					for o := range objects {
-						_, err := client.PutObject(ctx, repo, o.Path, strings.NewReader(o.Content), int64(len(o.Content)), minio.PutObjectOptions{})
+						_, err := clt.PutObject(ctx, repo, o.Path, strings.NewReader(o.Content), int64(len(o.Content)), minio.PutObjectOptions{})
 						if err != nil {
 							t.Errorf("minio.Client.PutObject(%s): %s", o.Path, err)
 							continue
 						}
 
-						download, err := client.GetObject(ctx, repo, o.Path, minio.GetObjectOptions{})
+						download, err := clt.GetObject(ctx, repo, o.Path, minio.GetObjectOptions{})
```

Comment:
based on the other code here it looks like we don't close the response body for this call. unless there is no need to close the response body for minio client GetObject and we need to remove the close from other places.